pkg/asset/installconfig/aws/instancetypes.go

```diff
@@ -0,0 +1,34 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/pkg/errors"
+)
+
+// InstanceType holds metadata for an instance type.
+type InstanceType struct {
+	DefaultVCpus int64
+	MemInMiB     int64
+}
+
+// instanceTypes retrieves a list of instance types for the given region.
+func instanceTypes(ctx context.Context, session *session.Session, region string) (map[string]InstanceType, error) {
+	client := ec2.New(session, aws.NewConfig().WithRegion(region))
+	resp, err := client.DescribeInstanceTypesWithContext(ctx, &ec2.DescribeInstanceTypesInput{})
```

Comment:
this came to light from a slack conversation. we should be using https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#EC2.DescribeInstanceTypesPagesWithContext  to make sure we list all the instance types.

Reply:
Converted to use Pages.